File: find_minimum_in_rotated_sorted_array.py
class Solution:
    # @param num, a list of integer
    # @return an integer
    def findMin(self, num):
        if not num:
            return -1
        minimum = 1 << 32
        lo, hi = 0, len(num)-1
        while lo < hi and num[lo] > num[hi]:
            mid = (lo+hi) // 2
            if num[mid] < num[hi]:
                hi = mid
            else:
                lo = mid + 1
        return num[lo]

File: java/FindMinimuminRotatedSortedArray.java
/**
 * Created by Robbert on 1/31/15.
 * Find Minimum in Rotated Sorted Array
 * https://oj.leetcode.com/problems/find-minimum-in-rotated-sorted-array/
 */
public class FindMinimuminRotatedSortedArray {
    public int findMinLinear(int[] num) {
        if (num == null || num.length == 0) {
            return 0;
        }
        int min = Integer.MAX_VALUE;
        for (int elem : num) {
            min = Math.min(elem, min);
        }
        return min;
    }
}
